profile/profile.js

```diff
@@ -0,0 +1,361 @@
+$(document).ready(function() {
+	loadProfileInformation();
+	loadAbilities();
+	loadShifts();
+
+	initializeEventListeners();
+});
+
+let loadProfileInformation = function() {
+	$.ajax({
+		url: "/server/profile/profile.php",
+		type: "GET",
+		dataType: "JSON",
+		data: {
+			action: 'getUserInformation'
+		},
+		cache: false,
+		success: function(response) {
+			$("#firstNameText").html(response.firstName);
+			$("#lastNameText").html(response.lastName);
+			$("#emailText").html(response.email);
+			$("#phoneNumberText").html(response.phoneNumber);
+		},
+		error: function(response) {
+			alert("Unable to load information. Please refresh the page in a few minutes.");
+		}
+	});
+};
+
+let loadAbilities = function() {
+	$.ajax({
+		url: "/server/profile/profile.php",
+		type: "GET",
+		dataType: "JSON",
+		data: {
+			action: 'getAbilities'
+		},
+		cache: false,
+		success: function(response) {
+			$('#abilitiesSelect').find('option').remove();
+			for (let i = 0; i < response.abilities.length; i++) {
+				let ability = response.abilities[i];
+				let modifiers = ability.has ? `data-userAbilityId=${ability.userAbilityId} selected` : '';
+				let option = $(`<option value=${ability.abilityId} ${modifiers}>${ability.name}</option>`);
+				$('#abilitiesSelect').append(option);
+			}
+			$('#abilitiesSelect').selectpicker();
+
+			$('#certificationsDiv').find('div').remove();
+			for (let i = 0; i < response.abilitiesRequiringVerification.length; i++) {
+				let ability = response.abilitiesRequiringVerification[i];
+				
+				let abilityDiv = $('<div></div>');
+				let abilityName = $('<span></span>').html(ability.name);
+				let abilityStatus = $('<i></i>').addClass(`${ability.has ? 'fa fa-check icon green-icon' : 'fa fa-times icon red-icon'}`);
+
+				abilityDiv.append(abilityName, abilityStatus);
+				$('#certificationsDiv').append(abilityDiv);
+			}
+		},
+		error: function(response) {
+			alert("Unable to load abilities. Please refresh the page in a few minutes.");
+		}
+	});
+};
+
+// Maps a siteId -> dates that site is open -> shifts for that date
+let shiftsMap = new Map();
+// Maps a siteId to the title of the site
+let sitesMap = new Map(); 
+
+let loadShifts = function() {
+	$.ajax({
+		url: "/server/profile/profile.php",
+		type: "GET",
+		dataType: "JSON",
+		data: {
+			action: 'getShifts'
+		},
+		cache: false,
+		success: function(response) {
+			for (let i = 0; i < response.shifts.length; i++) {
+				let shift = response.shifts[i];
+
+				// Create the shiftsMap
+				if (!sitesMap.has(shift.siteId)) sitesMap.set(shift.siteId, shift.title);
+				if (!shiftsMap.has(shift.siteId)) shiftsMap.set(shift.siteId, new Map());
+				
+				let startDateTime = new Date(shift.startTime + ' CST');
+				let endDateTime = new Date(shift.endTime + ' CST');
+
+				let date = startDateTime.toLocaleDateString();
```

Comment:
This will not work if the user is in a different time zone. You can debug this by changing your system timezone in your Windows 10 Date & Time settings. You might have to restart your browser as well.

Instead, I would recommend formatting the shift time into a string on the server-side that is passed to the client as something like ```shiftTime: "5:30 PM - 8:00 PM"``` rather than send a startTime and endTime to the client to handle.